### ISSUE TITLE
[FastPitch1.1/PyTorch] How to improve performance on Long sentence when the FastPitch is trained on short sentence dataset?

### ISSUE THREAD

**Issue by JohnHerry** [NVIDIA/DeepLearningExamples#1192]
Related to **FastPitch1.1/PyTorch** 

**Describe the bug**

My FastPitch model is trained with short sentence dataset, When we use it to synthesize long text sentence, the result audio will get smaller and smaller volume. Is there any suggestion to advance the FastPitch, Letting it synthesis long sentence without split when It was trained on short sentence dataset?

**To Reproduce**
1. Install '...'
2. Set "..."
2. Launch '...'

**Expected behavior**

**Environment**
Please provide at least:
* Container version (e.g. pytorch:19.05-py3):
* GPUs in the system: (e.g. 8x Tesla V100-SXM2-16GB):
* CUDA driver version (e.g. 418.67):

**Comment by martinvk1** [comment]
In my own experience you can't get good generalization to unseen input lengths, especially when the architecture is based on Transformers (LSTMs are actually better here). One idea is to just use longer sentences for training. If you don't have any, you can simply join together the utterances you have in order to create longer ones. Of course, with the quadratic complexity of the Transformers, this is going to take up a lot of VRAM. Fortunately, you can just reduce the batch_size and increase gradient accumulation.

**Comment by shivammehta25** [comment]
So the problem here lies with the positional embeddings in the Transformer models. I use relative positional embeddings to mitigate this problem but there are other works to fix this problem as well, AliBi is one such example: https://arxiv.org/pdf/2108.12409.pdf

In my code, I have changed from sinusoidal positional embeddings to relative positional embedding, if that is of any help you can find my source code here: 
[MultiHeadAttention with relative positional embeddings](https://github.com/shivammehta25/OverFlow/blob/9bd73e3f2e8402f28a49c7b22ea19de422d3fe1f/src/model/transformer.py#L81)
It has this `rel_attention: bool` parameter just pass true here and replace this class with the MultiHeadAttn class present in the fastpitch source code. It should mitigate the long sentence issues. 

Hope this helps! :)
